integration/spark/shared/src/main/java/io/openlineage/spark/agent/facets/builder/CustomEnvironmentFacetBuilder.java

```diff
@@ -0,0 +1,108 @@
+/*
+/* Copyright 2018-2023 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.spark.agent.facets.builder;
+
+import io.openlineage.spark.agent.facets.EnvironmentFacet;
+import io.openlineage.spark.api.CustomFacetBuilder;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.spark.scheduler.SparkListenerJobStart;
+
+/**
+ * {@link CustomFacetBuilder} that generates a {@link EnvironmentFacet} when using OpenLineage on
+ * Databricks.
+ */
+@Slf4j
+public class CustomEnvironmentFacetBuilder
+    extends CustomFacetBuilder<SparkListenerJobStart, EnvironmentFacet> {
+  private Map<String, Object> envProperties;
+
+  public static boolean isCustomEnvironmentVarCollectionEnabled() {
+    return System.getenv().containsKey("CUSTOM_VARS");
+  }
+
+  @Override
+  protected void build(
+      SparkListenerJobStart event, BiConsumer<String, ? super EnvironmentFacet> consumer) {
+    consumer.accept(
+        "environment-properties", new EnvironmentFacet(getCustomEnvironmentalAttributes(event)));
```

Comment:
I wonder if this won't override another `DatabricksEnvironmentFacetBuilder` that emits the same facet. Maybe there should be some merging there, or those facet builders be merged.

@pawel-big-lebowski what do you think?

Reply:
There will be an override I think...I will edit this class to merge with the custom vars instead.